components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/handler/UserSelfRegistrationHandler.java

```diff
@@ -422,11 +422,21 @@ protected void triggerNotification(User user, String type, String code, Property
      * @param eventName           Name of the event
      * @throws IdentityRecoveryException Error triggering notifications
      */
-    private void triggerNotification(User user, String notificationChannel, String code, Property[] props,
+    public void triggerNotification(User user, String notificationChannel, String code, Property[] props,
             String eventName) throws IdentityRecoveryException {
 
+        boolean emailOTPenabled = false;
```

Comment:
Take this logic out of triggernotification

Reply:
If we need to take this out of the TriggerNotification method then we need to pass this as an additional parameter to the method. If that is OK, I can take this out of the method and add an additional parameter for this method.